### DISCUSSION TITLE
打包出错，每次运行打包都要重新安装一次rust

### DISCUSSION THREAD

**Discussion by midpoint** [tw93/Pake#170] [Q&A]
u22-198:~/pake/Pake$ sudo pake https://weekly.tw93.fun --name Weekly --transparent
To build the Linux app, you need to install Rust and Linux package
See more in https://tauri.app/v1/guides/getting-started/prerequisites#installing

✔ We detected that you have not installed Rust. Install it now? … yes
⠦ Downloading Rustinfo: downloading installer
⠸ Downloading Rustinfo: profile set to 'default'
info: default host triple is x86_64-unknown-linux-gnu
warning: Updating existing toolchain, profile choice will be ignored
⠼ Downloading Rustinfo: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
⠹ Downloading Rustinfo: default toolchain set to 'stable-x86_64-unknown-linux-gnu'

  stable-x86_64-unknown-linux-gnu unchanged - rustc 1.65.0 (897e37553 2022-11-02)

Rust is installed now. Great!

To get started you may need to restart your current shell.
This would reload your PATH environment variable to include
Cargo's bin directory ($HOME/.cargo/bin).

To configure your current shell, run:
source "$HOME/.cargo/env"
✔ Downloading Rust
You have not provided an app icon, use the default icon.(use --icon option to assign an icon)

npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead

added 81 packages, and audited 235 packages in 21s

76 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> pake-cli@0.1.3 build
> npm run tauri build --release

> pake-cli@0.1.3 tauri
> tauri build

thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/interface/rust.rs:762:69
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: ENOENT: no such file or directory, copyfile '/usr/local/lib/node_modules/pake-cli/src-tauri/target/release/bundle/deb/Weekly_1.0.0_amd64.deb' -> '/home/mj/pake/Pake/Weekly.deb'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'copyfile',
  path: '/usr/local/lib/node_modules/pake-cli/src-tauri/target/release/bundle/deb/Weekly_1.0.0_amd64.deb',
  dest: '/home/mj/pake/Pake/Weekly.deb'
}

Node.js v18.12.1

**Comment by midpoint** [comment]
用普通权限运行打包命令，不会提示重新安装rust，提示信息如下：
u22-198:~/pake/Pake$ pake https://weekly.tw93.fun --name Weekly --transparent
To build the Linux app, you need to install Rust and Linux package
See more in https://tauri.app/v1/guides/getting-started/prerequisites#installing

You have not provided an app icon, use the default icon.(use --icon option to assign an icon)
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: EACCES: permission denied, open '/usr/local/lib/node_modules/pake-cli/src-tauri/tauri.linux.conf.json'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/usr/local/lib/node_modules/pake-cli/src-tauri/tauri.linux.conf.json'
}

Node.js v18.12.1

**Reply by Tlntin** [reply]
你这个很明显是权限问题了呀。
/usr/local/lib/node_modules/pake-cl
这个路径是系统路径，也就是说，你用普通用户的时候，就算你已经用了普通用户权限，优先用的系统路径的pake-cli，而pake-cli使用的时候会修改这个路径下的文件，所以就会导致无权限问题。
解决办法：
卸载系统用户的pake-cli
```bash
sudo npm uninstall -g pake-cli
```
然后参考之前的帖子将npm包装到普通权限去：
https://gist.github.com/Giancarlos/d087f8a9e6516716da98ad0c0f5a8f58
再普通用户权限安装pake-cli
```bash
npm install -g pake-cli
```
文档写的很清楚啦，没权限的解决办法。已经回复了很多次了[/笑哭]
![image](https://user-images.githubusercontent.com/28218658/207754490-7e9273f8-51b9-45b3-ba6f-06c4ba4f19d3.png)

**Comment by midpoint** [comment]
感谢！  果然是权限的问题
习惯用sudo安装了，这个要用普通用户
卸载后，重新用普通用户安装一遍就可以了。
再次感谢！

**Reply by Tlntin** [reply]
用sudo也是可以的，只不过rust也要用sudo apt 来安装（这个备注已经添加到文档了），普通人建议还是普通权限更佳。
![image](https://user-images.githubusercontent.com/28218658/207765262-5ec21d2d-b60c-45e7-927c-c6735c007941.png)
